resource-aggregate/service/grpcApi.go

```diff
@@ -155,6 +155,9 @@ func (r RequestHandler) UnpublishResourceLinks(ctx context.Context, request *com
 		log.Errorf("cannot publish resource links unpublished events: %v", err)
 	}
 	auditContext := commands.NewAuditContext(owner, "")
+
+	r.UnpublishResources(ctx, request, owner, events)
```

Comment:
This from flow doesn't make sense. You unpublished resource links on L148. What are you doing here when you are already creating a response? Why another request handler? It's not readable. Mainly because you call it unpublish resources, what has nothing to do with resource state. Call it for example CleanupResourceState. Including go file name, functions, everything.

Reply:
Fixed